runtime/src/snapshot_utils.rs

```diff
@@ -2120,12 +2167,23 @@ pub fn verify_snapshot_archive<P, Q, R>(
     if let VerifyBank::NonDeterministic(slot) = verify_bank {
         // file contents may be different, but deserialized structs should be equal
         let slot = slot.to_string();
+        let snapshot_slot_dir = snapshots_to_verify.as_ref().join(&slot);
         let p1 = snapshots_to_verify.as_ref().join(&slot).join(&slot);
         let p2 = unpacked_snapshots.join(&slot).join(&slot);
 
         assert!(crate::serde_snapshot::compare_two_serialized_banks(&p1, &p2).unwrap());
         std::fs::remove_file(p1).unwrap();
         std::fs::remove_file(p2).unwrap();
+
+        // Remove the new accounts/ to be consistent with the
+        // old archive structure.
+        let accounts_path = snapshot_slot_dir.join("accounts");
+        if accounts_path.is_dir() {
+            // Do not use the async move_and_async_delete_path because the assert below
+            // requires the job to be done.
+            // This is for test only, so the performance is not an issue.
+            std::fs::remove_dir_all(accounts_path).unwrap();
+        }
```

Comment:
Todo: will move this out of if VerifyBank::NonDeterministic.  Will check if it can be combined with the accounts check below.

Reply:
Moved out of if VerifyBank::NonDeterministic.
No need to combine this deletion with the next accounts check.  The deletion is just to remove the side effects of the new extra files.  The existing accounts check is not affected.

Reply:
Actually, i ran into compilation problem after moving that out of if VerifyBank::NonDeterministic, because the slot is derived from that.  The slot and snapshot_slot_dir are in the scope of the if block.  So I have to revert the changes.  @brooksprumo, please let me know if it is OK with you.

Reply:
These comments were added:

        // Remove the new accounts/ hard-link directory to be consistent with the
        // old archive structure, so that the directory comparison checking in the test
        // passes.

Reply:
Replaced with your version of the comments.